irods/src/main/java/ch/cyberduck/core/irods/IRODSSession.java

```diff
@@ -49,23 +48,24 @@
 import ch.cyberduck.core.ssl.X509TrustManager;
 import ch.cyberduck.core.threading.CancelCallback;
 
+import ch.cyberduck.core.worker.DefaultExceptionMappingService;
+
 import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.irods.jargon.core.connection.AuthScheme;
-import org.irods.jargon.core.connection.IRODSAccount;
-import org.irods.jargon.core.connection.SettableJargonProperties;
-import org.irods.jargon.core.connection.auth.AuthResponse;
-import org.irods.jargon.core.exception.JargonException;
-import org.irods.jargon.core.pub.IRODSAccessObjectFactory;
-import org.irods.jargon.core.pub.IRODSFileSystem;
-import org.irods.jargon.core.pub.IRODSFileSystemAO;
-
-import java.net.URI;
-import java.net.URISyntaxException;
+import org.irods.irods4j.high_level.connection.IRODSConnection;
+import org.irods.irods4j.high_level.connection.QualifiedUsername;
+import org.irods.irods4j.low_level.api.IRODSApi;
+import org.irods.irods4j.low_level.api.IRODSException;
+
 import java.text.MessageFormat;
 
-public class IRODSSession extends SSLSession<IRODSFileSystemAO> {
+public class IRODSSession extends SSLSession<IRODSConnection> {
+
+    static {
+        IRODSApi.setApplicationName(new PreferencesUseragentProvider().get());
```

Comment:
This change revealed a bug in the iRODS server. See the following for details.
- https://github.com/irods/irods/issues/8733

While the bug doesn't affect Cyberduck's ability to interact with the iRODS server, it does affect an administrator's ability to see connections from Cyberduck.

So, I'm going to hard-code the application name until the issue is resolved. At the very least, admins will be able to see connections and make decisions.

I can include a TODO comment or open an issue as a reminder if that sounds helpful.